packages/frontend/core/src/modules/doc-display-meta/services/doc-display-meta.ts

```diff
@@ -143,6 +154,41 @@ export class DocDisplayMetaService extends Service {
     });
   }
 
+  title(idOrMeta: string | DocMeta, options?: DocDisplayTitleOptions) {
+    const docId = typeof idOrMeta === 'string' ? idOrMeta : idOrMeta.id;
+    const docTitle =
+      typeof idOrMeta === 'string'
+        ? this.docsService.list.doc(idOrMeta).title
+        : idOrMeta.title;
+
+    const journalDateString = this.journalService.getJournalDate(docId);
+
+    // journal
+    if (journalDateString) {
+      return i18nTime(journalDateString, { absolute: { accuracy: 'day' } });
+    }
+
+    if (options?.originalTitle) return options.originalTitle;
+
+    // empty title
+    if (!docTitle) return { i18nKey: 'Untitled' } as const;
+
+    // reference
+    if (options?.reference) return docTitle;
+
+    // check emoji
+    const enableEmojiIcon =
+      this.featureFlagService.flags.enable_emoji_doc_icon.value &&
+      options?.enableEmojiIcon !== false;
+    if (enableEmojiIcon) {
+      const { rest } = extractEmojiIcon(docTitle);
+      return rest;
+    }
+
+    // default
+    return docTitle;
```

Comment:
maybe extract this logic shared with `title$`?

Reply:
there are some subtle different between this and title$.. I am not sure if there is a good way to share the logic instead of having two almost identical implementations.